controllers/storagecluster/platform_detection.go

```diff
@@ -63,6 +91,39 @@ func avoidObjectStore(p configv1.PlatformType) bool {
 	return false
 }
 
+func getActualIBMPlatformType(c client.Client) (configv1.PlatformType, error) {
+	isSecretPresent, secErr := IsCosSecretPresent(c)
+	if secErr != nil {
+		return "", fmt.Errorf("Error checking COS secret in IBMCloud: %v", secErr)
+	}
+	if isSecretPresent {
+		// IsIBMCloud and COS Secret present.
+		// return new CloudProviderType
+		return IBMCloudCosPlatformType, nil
+	}
+	//COS secret is not present in IBMCloudPlatform
+	return configv1.IBMCloudPlatformType, nil
+}
+
+// IsCosSecretPresent checks for ibm-cos-cred secret in the concerned namespace
+// if platform is IBMCloud, enable CephObjectStore only if ibm-cloud-cos-creds secret is not present
+// in the target namespace
+func IsCosSecretPresent(c client.Client) (bool, error) {
+	ns, nsErr := util.GetWatchNamespace()
```

Comment:
Add a `TODO:` comment above this line to indicate we should try and not do this if possible.